File: ase/samples/word.js

var quit_flag = false;
var cb = {
  'Quit' : function() {
    print("QUIT");
    quit_flag = true;
  }
};
var word = ASE.COM.CreateObject("Word.Application");
ASE.COM.Connect(word, cb);
word.Visible = true;
while (!quit_flag) {
  ASE.COM.DispatchMessage(1000);
  print("WAIT");
}
print("DONE");

File: ase/samples/word.lua

local quit_flag = false;
local cb = {
  Quit = function()
    print("QUIT")
    quit_flag = true
  end,
};
local word = ASE.COM.CreateObject("Word.Application");
ASE.COM.Connect(word, cb);
word.Visible = true;
while (not quit_flag) do
  ASE.COM.DispatchMessage(1000);
  print("WAIT");
end
print("DONE");

